CONTRIBUTING.md

```diff
@@ -0,0 +1,289 @@
+Here is a more detailed example of the commit message convention:
+
+```
+    fix #ISSUE Prefer short (50 chars) summary of changes
+
+    More detailed explanatory text, if necessary. Wrap it to about 72
+    characters or so. In some contexts, the first line is treated as the
+    subject of an email and the rest of the text as the body. The blank
+    line separating the summary from the body is critical (unless you omit
+    the body entirely); tools like rebase can get confused if you run the
+    two together.
+
+    Further paragraphs come after blank lines.
+
+     - Bullet points are okay, too
+
+     - Typically a hyphen or asterisk is used for the bullet, preceded by a
+       single space, with blank lines in between, but conventions vary here
+
+    Related Issues: #1234, gh-1235
+```
+
+
+1. Use imperative statements in the subject line, e.g. "Fix broken Javadoc link".
+1. Begin the DESCRIPTION with a capitalized verb, e.g. "Add, Prune, Fix,
+    Introduce, Avoid, etc."
+1. Prefix the subject line with "fix #XXX " if the commit is fixing issue XXX (or
+    replace "fix" with "see" if it is only related to an issue without closing it)
+1. Do not end the subject line with a period.
+1. Restrict the subject line to 72 characters or less.
+1. Wrap lines in the body at 72 characters or less.
+1. In the body of the commit message, explain how things worked before this
+    commit, what has changed, and how things work now.
+
+#### Using real names
+Please configure git to use your real first and last name for any commits you
+intend to submit as pull requests. For example, this is not acceptable:
+
+    Author: Nickname <user@mail.com>
+
+Rather, please include your first and last name, properly capitalized, as
+submitted against the Spring Individual Contributor License Agreement (ICLA):
+
+    Author: First Last <user@mail.com>
+
+This helps ensure traceability against the ICLA and also goes a long way to
+ensuring useful output from tools like `git shortlog` and others.
+
+You can configure this via the account admin area in GitHub (useful for
+fork-and-edit cases); _globally_ on your machine with
+
+    git config --global user.name "First Last"
+    git config --global user.email user@mail.com
+
+or _locally_ for the `reactor-core` repository only by omitting the
+'--global' flag:
+
+    cd reactor-core
+    git config user.name "First Last"
+    git config user.email user@mail.com
+    
+### Importing and following the Reactor Code Style
+
+Using your IDE code style support:
+
+Eclipse users can import the content of `$PROJECT_DIR/codequality/eclipse`
+
+IntelliJ users can import the xml from `$PROJECT_DIR/codequality/idea`.
+
+IntelliJ users: If your IDE doesn't support import of xml files
+(versions prior to 2016.1) you can copy manually into
+`$HOME/.IntelliJIdea{version}/codestyles`
+
+#### Add Apache license header to all new classes
+
+```java
+/*
+ * Copyright (c) 2011-2019 Pivotal Software Inc, All Rights Reserved.
```

Comment:
Latest info form the legal folks was that we can (and probably should) use `-Present`, to avoid updating it every year

Reply:
fixed in 11e060d